arelle/plugin/validate/EDINET/rules/gfm.py

```diff
@@ -670,3 +670,32 @@ def rule_gfm_1_3_8(
             msg=_("The submitter-specific taxonomy has an embedded linkbase."),
             modelObject=embeddedElements
         )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_5_6(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.5.6] The length of a label must be less than 511 characters unless it its role is documentation.
+    """
+    labelRelationshipSet = val.modelXbrl.relationshipSet(XbrlConst.conceptLabel)
+    if labelRelationshipSet is not None:
+        for concept in val.modelXbrl.qnameConcepts.values():
+            labelRels = labelRelationshipSet.fromModelObject(concept)
+            for rel in labelRels:
+                label = rel.toModelObject
+                if label.role != XbrlConst.documentationLabel and label.viewText() is not None and len(label.viewText()) > 511:
+                    yield Validation.warning(
+                        codes='EDINET.EC5700W.GFM.1.5.6',
+                        msg=_("The concept of \"{concept}\" has a label classified as \"{role}\" that is longer than 511 characters: {label}".format(
+                            concept=concept.qname, role=label.role, label=label.viewText()
+                        )),
```

Comment:
This approach to string interpolation is not consistent with the rest of our validation messages.